les/client_handler.go

```diff
@@ -353,6 +398,25 @@ func (h *clientHandler) handleMsg(p *peer) error {
 		p.fcServer.ReceivedReply(resp.ReqID, resp.BV)
 		p.Log().Trace("Setting peer etherbase", "etherbase", resp.Etherbase, "Peer ID", p.ID)
 		p.SetEtherbase(resp.Etherbase)
+
+	case GatewayFeeMsg:
+		var resp struct {
+			ReqID, BV uint64
+			Data      GatewayFeeResps
+		}
+
+		if err := msg.Decode(&resp); err != nil {
+			return errResp(ErrDecode, "msg %v: %v", msg, err)
+		}
+
+		p.fcServer.ReceivedReply(resp.ReqID, resp.BV)
+
+		currGatewayFee := big.NewInt(int64(resp.Data.GatewayFee))
+		currEtherbase := resp.Data.Etherbase
+		nodeID := resp.Data.NodeID
```

Comment:
As we discussed, it is better to get this information form the peer object (`p`) here.